From f7762cf56ba4aa9384a4d05d32e1fc8825730eda Mon Sep 17 00:00:00 2001
From: tnmy-cyngn <tlnu@cyngn.com>
Date: Thu, 23 Jun 2016 09:36:58 +0530
Subject: [PATCH 12/16] Messaging app crashes after a few MMS

In some scenarios, data not bound when SIM state
change local broadcasts are received. See stacktrace
for more info. Protect against/skip call to update()
in such scenarios.

Change-Id: Iee7775a74d22808cb262e8973ca70e5e1f0acfc9
Bug-Id: CYNGNOS-3062, KIPPER-718
---
 .../messaging/ui/conversation/ConversationFragment.java        | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/com/android/messaging/ui/conversation/ConversationFragment.java b/src/com/android/messaging/ui/conversation/ConversationFragment.java
index a6a191a..419ad5d 100644
--- a/src/com/android/messaging/ui/conversation/ConversationFragment.java
+++ b/src/com/android/messaging/ui/conversation/ConversationFragment.java
@@ -191,7 +191,8 @@ public class ConversationFragment extends Fragment implements ConversationDataLi
                     intent.getStringExtra(UIIntents.UI_INTENT_EXTRA_CONVERSATION_SELF_ID);
             Assert.notNull(conversationId);
             Assert.notNull(selfId);
-            if (TextUtils.equals(mBinding.getData().getConversationId(), conversationId)) {
+            if (isBound() && TextUtils
+                    .equals(mBinding.getData().getConversationId(), conversationId)) {
                 mComposeMessageView.updateConversationSelfIdOnExternalChange(selfId);
             }
         }
-- 
2.17.1

